golem-worker-executor-base/tests/guest_languages3.rs

```diff
@@ -55,9 +55,15 @@ async fn javascript_example_1(
 
     let end = chrono::Utc::now().timestamp_millis() as u64;
 
-    tokio::time::sleep(Duration::from_secs(5)).await;
     let mut events = vec![];
-    rx.recv_many(&mut events, 100).await;
+    let start_time = Instant::now();
+    while events.len() < 2 && start_time.elapsed() < Duration::from_secs(5) {
```

Comment:
that makes sense

Reply:
Thinking more about it, it's harder to come up with a good timeout on the whole test, because things like the compilation time of the WASM are hard to predict (how slow it is on CI). 
Saying "after the invocation is done, we wait at most 5 seconds for the async log fetching to return some things" feels easier to work with to me.